[Bug]: @remix-validated-form/with-zod does not compile with typescript 4.8.x

### Which packages are impacted?

- [ ] `remix-validated-form`
- [X] `@remix-validated-form/with-zod`
- [ ] `@remix-validated-form/with-yup`
- [ ] `zod-form-data`

### What version of these packages are you using?

- `@remix-validated-form/with-zod: 2.0.1`
- `zod: 3.19.1`

### Please provide a link to a minimal reproduction of the issue.

(unable to reproduce on codesandbox - compiling hangs)

### Steps to Reproduce the Bug or Issue

Compilation error: `@remix-validated-form/with-zod/build/index.d.ts(6,62): error TS2344: Type 'U' does not satisfy the constraint 'ZodTypeDef'.`

### Expected behavior

Should compile with no errors.

### Screenshots or Videos

_No response_

### Platform

OS: [node on osx]
Browser: [none]
Version: [none]

### Additional context

_No response_

--- a/packages/with-zod/src/index.ts
+++ b/packages/with-zod/src/index.ts
@@ -23,7 +23,7 @@ function pathToString(array: (string | number)[]): string {
 /**
  * Create a validator using a `zod` schema.
  */
-export function withZod<T, U>(
+export function withZod<T, U extends z.ZodTypeDef>(
   zodSchema: z.Schema<T, U, unknown>
 ): Validator<T> {
   return createValidator({

